application/dash_application/views/call_ready_rna.py

```diff
@@ -0,0 +1,428 @@
+from collections import defaultdict
+import logging
+
+import dash_html_components as html
+import dash_core_components as core
+from dash.dependencies import Input, Output, State
+
+import gsiqcetl.column
+import pinery
+from ..dash_id import init_ids
+from ..utility.plot_builder import *
+from ..utility.table_builder import table_tabs, cutoff_table_data_merged
+from ..utility import df_manipulation as util
+from ..utility import sidebar_utils
+from ..utility import log_utils
+
+logger = logging.getLogger(__name__)
+
+page_name = 'call-ready-rna'
+title = "Call-Ready RNA"
+
+ids = init_ids([
+    # Buttons
+    'update-button',
+
+    # Sidebar controls
+    'all-projects',
+    'projects-list',
+    'all-tissue-preps',
+    'tissue-preps-list',
+    'all-sample-types',
+    'sample-types-list',
+    'first-sort',
+    'second-sort',
+    'colour-by',
+    'shape-by',
+    'search-sample',
+    'show-data-labels',
+    'show-all-data-labels',
+    'pf-cutoff',
+    'rrna-contam-cutoff',
+
+    # Graphs
+    'total-reads',
+    'unique-reads',
+    'five-to-three-bias',
+    'correct-read-strand',
+    'coding',
+    'rrna-contam',
+
+    # Tables
+    'failed-samples',
+    'data-table'
+])
+
+RNASEQQC_COL = gsiqcetl.column.RnaSeqQc2MergedColumn
+PINERY_COL = pinery.column.SampleProvenanceColumn
+
+
+def dataversion():
+    return DATAVERSION
+
+
+special_cols = {
+    "Total Reads (Passed Filter)": "Total reads passed filter",
+    "Unique Reads (PF)": "Unique reads",
+    "File SWID RNAseqQC": "File SWID RNAseqQC",
+    "% rRNA Contamination": "Percent rRNA Contamination"
+}
+
+def get_merged_rna_data():
+    """
+    Join together the RNAseqQC and Pinery dataframes
+    """
+    # Sample metadata from Pinery
+    pinery_samples = util.get_pinery_merged_samples()
+    pinery_samples = util.filter_by_library_design(pinery_samples, util.rna_lib_designs)
+
+    rnaseqqc_df = util.get_rnaseqqc_merged()
+    rnaseqqc_df = util.filter_by_library_design(rnaseqqc_df, util.rna_lib_designs, RNASEQQC_COL.LibraryDesign)
+
+    rnaseqqc_df[special_cols["Total Reads (Passed Filter)"]] = round(
+        rnaseqqc_df[RNASEQQC_COL.TotalReads] / 1e6, 3)
+    rnaseqqc_df[special_cols["Unique Reads (PF)"]] = round(
+        rnaseqqc_df[RNASEQQC_COL.UniqueReads] / 1e6, 3)
+    rnaseqqc_df[special_cols["% rRNA Contamination"]] = round(
+        (rnaseqqc_df[RNASEQQC_COL.RRnaContaminationInTotal] / rnaseqqc_df[
+            RNASEQQC_COL.TotalReads]) * 100, 2)
```

Comment:
If this view will use rnaseqqc (and not rnaseqqc2) then this can be replaced with `rRNAContaminationreadsaligned` (which is used by Single Lane RNA-Seq).

Reply:
As this uses rnaseqqc2, we don't have that field available.